sysconfig/cloudinit.go

```diff
@@ -263,6 +263,68 @@ func cloudDatasourcesInUse(configFile string) (*cloudDatasourcesInUseResult, err
 	return res, nil
 }
 
+// cloudDatasourcesInUseForDir considers all files in a directory as individual
+// cloud-init config files, and analyzes all datasources in use for each file
+// and returns their union. It does not distinguish between mentioned,
+// explicitly allowed, or explicitly disallowed, but it does follow cloud-init's
+// logic for determining the overwriting of properties such that if a lexcially
+// earlier file sets datasource_list: [], then a later file sets
+// datasource_list: [foo], then foo is used instead and the explicit disallowing
+// is ignored/overwritten, but if no other file sets datasource_list, then it is
+// treated as if the config explicitly disallows no datasources.
+func cloudDatasourcesInUseForDir(dir string) (*cloudDatasourcesInUseResult, error) {
+
+	files, err := filepath.Glob(filepath.Join(dir, "*"))
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: validate that this is the same order in which files are processed
```

Comment:
Is this a TODO you plan to address in this branch? Or is this just a matter of manually checking that cloud-init code follows the same algorithm?

Reply:
sorry yes this was a TODO for myself I forgot to delete from this branch